### PR TITLE
Add Scale CD for hipBLASLt

### PR DESCRIPTION
**Pull request by KKyang**

### CODE REVIEW

**Comment by KKyang** [comment]
gfx940 local test passed.

File: clients/common/cblas_interface.cpp [hunk 1]
```diff
@@ -93,8 +95,16 @@ void cblas_gemm<hip_bfloat16, hip_bfloat16, hip_bfloat16, float>(hipblasOperatio
                 C_float,
                 ldc);
 
+    if(doScaleD)
+    {
+    for(size_t i = 0; i < sizeC; i++)
```
Comment (jichangjichang) [inline clients/common/cblas_interface.cpp:100]: Did you run clang-format?

File: clients/include/testing_matmul.hpp [hunk 2]
```diff
@@ -90,11 +96,17 @@ void epilogue_func(int64_t m,
                    To*     out,
                    Tc*     out_raw,
                    Tc*     e,
+                   bool    doScaleD,
```
Comment (jichangjichang) [inline clients/include/testing_matmul.hpp:99]: doScaleD and doScaleE are unneccesary. scaleD and scaleE would be set as 1 in disabling scaleD or scaleE cases.

File: library/include/hipblaslt.h [hunk 3]
```diff
@@ -185,10 +185,13 @@ typedef enum {
...
-  HIPBLASLT_MATMUL_DESC_EPILOGUE_AUX_LD = 8,            /**<The leading dimension of the epilogue auxiliary buffer pointer in the device memory. Data Type:int64_t */
-  HIPBLASLT_MATMUL_DESC_EPILOGUE_AUX_BATCH_STRIDE = 9,  /**<The batch stride of the epilogue auxiliary buffer pointer in the device memory. Data Type:int64_t */
-  HIPBLASLT_MATMUL_DESC_POINTER_MODE = 10,              /**<Specifies alpha and beta are passed by reference, whether they are scalars on the host or on the device, or device vectors. Default value is: HIPBLASLT_POINTER_MODE_HOST (i.e., on the host). Data Type: int32_t based on hipblasLtPointerMode_t*/
+  HIPBLASLT_MATMUL_DESC_C_SCALE_POINTER = 7,            /**<Equivalent to HIPBLASLT_MATMUL_DESC_A_SCALE_POINTER for matrix C. Default value: NULL Type: void* /const void* */
+  HIPBLASLT_MATMUL_DESC_D_SCALE_POINTER = 8,            /**<Equivalent to HIPBLASLT_MATMUL_DESC_A_SCALE_POINTER for matrix D. Default value: NULL Type: void* /const void* */
+  HIPBLASLT_MATMUL_DESC_AUX_SCALE_POINTER = 9,          /**<Equivalent to HIPBLASLT_MATMUL_DESC_A_SCALE_POINTER for matrix AUX. Default value: NULL Type: void* /const void* */
```
Comment (jichangjichang) [inline library/include/hipblaslt.h:190]: should be HIPBLASLT_MATMUL_DESC_EPILOGUE_AUX_SCALE_POINTER

Comment (KKyang) [reply library/include/hipblaslt.h]: fixed

File: clients/include/testing_matmul.hpp (hunk 2 above)
Comment (KKyang) [reply clients/include/testing_matmul.hpp]: fixed

File: clients/common/cblas_interface.cpp (hunk 1 above)
Comment (KKyang) [reply clients/common/cblas_interface.cpp]: fixed

File: clients/gtest/matmul_gtest.yaml [hunk 4]
```diff
@@ -917,4 +917,25 @@ Tests:
...
+  function:
+    matmul: *real_mix_precisions_fp8
+  M: [127, 129]
+  N: [127, 129]
+  K: [127, 129]
+  transA_transB: *transA_transB_range
+  alpha: 1
+  beta: [ 0.0, 2.0 ]
+  scaleA: [1]
+  scaleB: [1]
```
Comment (jichangjichang) [inline clients/gtest/matmul_gtest.yaml:932]: Please also test no scaleA(Or B,C,D case) in supported kernel

Comment (KKyang) [reply clients/gtest/matmul_gtest.yaml]: Fixed

**Review by jichangjichang — APPROVED** [review]
LGTM

**Review by aazz44ss — APPROVED** [review]

**Pull request merged**